android/src/main/kotlin/com/devcycle/devcycle_flutter_client_sdk/DevCycleFlutterClientSdkPlugin.kt

```diff
@@ -340,29 +343,40 @@ class DevCycleFlutterClientSdkPlugin: FlutterPlugin, MethodCallHandler {
     variableAsMap["id"] = variable.id
     variableAsMap["key"] = variable.key
     variableAsMap["type"] = variable.type.toString()
-    variableAsMap["value"] = if (value is JSONObject) value.toMap() else value
+    variableAsMap["value"] = value
     variableAsMap["evalReason"] = variable.evalReason
 
+    if (value is JSONObject) variableAsMap["value"] = value.toMap()
+    if (value is JSONArray) variableAsMap["value"] = value.toMap()
+
     return variableAsMap
   }
 
   private fun <T> watchForVariableUpdate(variable: Variable<T>, variableUpdateMap: MutableMap<String, Variable<T>>) {
     if (variable.key !in variableUpdateMap) {
       variable.onUpdate { result: Variable<T> ->
-        callFlutter("variableUpdated", variableToMap(result))
+        var value = result.value
+        val args = mutableMapOf(
+          "key" to result.key,
+          "value" to value
+        )
+        if (value is JSONObject) args["value"] = value.toMap()
+        if (value is JSONArray) args["value"] = value.toMap()
+        callFlutter("variableUpdated", args)
```

Comment:
this change isn't necessary anymore, it can be left as `callFlutter("variableUpdated", variableToMap(result))`